aptos-move/framework/aptos-experimental/sources/trading/order_book/order_book.move

```diff
@@ -170,34 +166,41 @@ module aptos_experimental::order_book {
         let order =
             new_order(
                 order_id,
-                unique_priority_idx,
+                ascending_idx,
                 order_req.price,
                 order_req.orig_size,
                 order_req.remaining_size,
-                order_req.is_buy,
+                order_req.is_bid,
                 order_req.trigger_condition,
                 order_req.metadata
             );
         self.orders.add(order_id, new_order_with_state(order, true));
         self.active_orders.place_maker_order(
             order_id,
             order_req.price,
-            unique_priority_idx,
+            ascending_idx,
             order_req.remaining_size,
-            order_req.is_buy
+            order_req.is_bid
         );
     }
 
     /// Reinserts a maker order to the order book. This is used when the order is removed from the order book
     /// but the clearinghouse fails to settle all or part of the order. If the order doesn't exist in the order book,
     /// it is added to the order book, if it exists, it's size is updated.
     public fun reinsert_maker_order<M: store + copy + drop>(
-        self: &mut OrderBook<M>,
-        order_req: OrderRequest<M>,
-        unique_priority_idx: UniqueIdxType
+        self: &mut OrderBook<M>, order_req: OrderRequest<M>, original_order: Order<M>
     ) {
-        assert!(order_req.trigger_condition.is_none(), E_NOT_ACTIVE_ORDER);
         let order_id = new_order_id_type(order_req.account, order_req.account_order_id);
+
+        assert!(&original_order.get_order_id() == &order_id, E_REINSERT_ORDER_MISMATCH);
+        assert!(
+            original_order.get_orig_size() >= order_req.orig_size,
```

Comment:
You need to check the original size is equal and the remaining size should be greater.